### ISSUE TITLE
Change documentation to Python 3

### ISSUE THREAD

**Issue by tigerhawkvok** [simplejson/simplejson#252]
Given that Python 2 is EOL-ing in less than six months, the documentation shouldn't be Py2 centric.

**Comment by etrepum** [comment]
It would be helpful if you could identify the parts of the documentation that are "Py2 centric". I'm not aware of any examples that are not compatible with both. A PR that demonstrates what you expect to see would be even better!

**Comment by tigerhawkvok** [comment]
Documentation intro:

> simplejson exposes an API familiar to users of the standard library marshal and pickle modules. It is the externally maintained version of the json library contained in Python 2.6, but maintains compatibility with Python 2.5 and (currently) has significant performance advantages, even without using the optional C extension for speedups. simplejson is also supported on Python 3.3+.

Explicitly declaring Unicode strings, eg:

```python
>>> import simplejson as json
>>> obj = [u'foo', {u'bar': [u'baz', None, 1.0, 2]}]
>>> json.loads('["foo", {"bar":["baz", null, 1.0, 2]}]') == obj
True
>>> json.loads('"\\"foo\\bar"') == u'"foo\x08ar'
True
>>> from simplejson.compat import StringIO
>>> io = StringIO('["streaming API"]')
>>> json.load(io)[0] == 'streaming API'
True
```

Unnecessary qualifier under "Basic Usage" -> `dump`:

> The simplejson module will produce str objects in Python 3, not bytes objects. Therefore, fp.write() must support str input.

There are actually a whole bunch of references to class `str` vs class `unicode`, which is *no difference* in Python 3 and just confusing.

I created a PR (#254) though I expect that you guys will substantially want to change it for under-the-hood accuracy, but the point is given that the target audience is Py3 developers there's no need to flag unicode strings, discuss coercion-to-unicode prominently (I prefaced those parts with "In Python 2" to allow devs to skip the whole block of text as irrelevant), etc.

**Comment by etrepum** [comment]
Resolved by #257
